docs/Technical-Documentation/tech-memos/feature-toggle-system.md

```diff
@@ -0,0 +1,620 @@
+            logger.debug(f"Cache TOUCH '{full_key}' (timeout={timeout})")
+            return result
+        except Exception as e:
+            logger.exception(f"Cache TOUCH error for key '{key}'")
+            return False
+
+
+# Convenience instances for common use cases
+default_cache = CacheService()
+feature_cache = CacheService(key_prefix='feature_flag')
+```
+
+This generic approach is initially more complex than what the feature flag system requires. However, it lays the foundation
+to grow the caching architecture without potentially more costly re-factors.
+
+#### Automated Cache Updates
+
+Feature flags will be primarily managed via the admin console. When a feature flag is created, updated, or deleted the cache becomes
+invalid and requires an appropriate update. This can be easily handled with signals. See an example below of a signal receiver for the
+`post_save` signal. The same pattern can be applied to the `post_delete` signal.
+
+**Location**: `tdrs-backend/tdpservice/core/signals.py`
+
+```python
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from .models import FeatureFlag
+from .services import feature_cache
+import logging
+
+logger = logging.getLogger(__name__)
+
+@receiver([post_save], sender=FeatureFlag)
+def update_feature_flag_cache(sender, instance, **kwargs):
+    """Update feature flag in cache when feature flags are modified."""
+    logger.info(
+        f"Feature flag '{instance.feature_name}' saved. "
+        f"Updating cache. New state: {instance.enabled}"
+    )
+
+    value = {
+        'enabled': instance.enabled,
+        'config': instance.config
+    }
+
+    feature_cache.set(instance.feature_name, value)
+```
+
+
+### Feature Flag Auditing
+Because the feature flag system allows changes to be made that will affect production that do not go through a formal review process,
+having a method to audit and rollback those changes is a necessity. The TDP backend has a model, `ChangeRequestAuditLog`, that is used
+for a similar purpose. However, the `ChangeRequestAuditLog` interface is tightly coupled to user change requests and does not allow an admin
+to rollback a change based on the change logged in the audit log. This is a common requirement for applications and is well supported by
+existing tools. This technical memorandum recommends implementing either `django-simple-history` or `django-reversion` to provide audit logging,
+robust versioning and rollback capabilities for `FeatureFlag`s and any other model in the future that requires it. The caveat here is that
+the TDP backend is running off of Django v3.2 and these packages require Django >v4.2. With that in mind this technical memorandum suggests
+the Dev team takes the time to upgrade the backend dependencies to support Django >v4.2. If this cannot be accommodated then the Dev team should
+investigate other options for audit logging and versioning.
+
+
+### Serialization and View(s)
```

Comment:
the view returns a plain dict of {feature_name: {enabled, config}} and doesn’t use FeatureFlagSerializer in the sample. Can't we use the serializer instead?